tests/document/indexes.py

```diff
@@ -792,6 +792,27 @@ class User(Document):
 
         User.drop_collection()
 
+    def test_unique_and_primary_create(self):
+        """Create a new record with a duplicate primary key
+        throws an exception
+        """
+
+        class User(Document):
+            name = StringField(primary_key=True)
+            password = StringField()
+
+        User.drop_collection()
+
+        User.objects.create(name='huangz', password='secret')
+        def duplicate_primary_key():
+            User.objects.create(name='huangz', password='secret2')
+        self.assertRaises(NotUniqueError, duplicate_primary_key)
+
+        self.assertEqual(User.objects.count(), 1)
+        self.assertEqual(User.objects.get().password, 'secret')
+
+        User.drop_collection()
```

Comment:
You can remove this line. The rule of thumb we use in this repo is that all unit tests should call `drop_collection` immediately after defining a document (as you do above), so there's no need for this redundant call.